docs/doc/90-contributing/01-rfcs/20230307-stage-with-schema.md

```diff
@@ -0,0 +1,102 @@
+---
+title: Stage With Schema
+---
+
+## Summary
+
+stage support `schema` option.
+
+## Motivation
+
+Currently, when query/copy/insert data from files, we obtain the schema by 2 ways:
+
+1. use the schema of the dest table (used in copy/insert).
+2. use the schema of inferred from data (used in stage_table_function).
+
+To support "transform during load", the first way can not be used.
+
+Inferred schema, although convenient, has many drawbacks (detailed in 'drawbacks of infer_schema' section below).
+
+## Guide-level explanation
+
+basic form:
+
+```sql
+SELECT c1, c2 + c1, trim(c2) from @my_stage(schema=(c1 int, c2 float, c3 string))
+```
+
+it can be used to support "transform during load".
+
+```sql
+copy into my_table from (
+    SELECT c1, c2 + c1, trim(c2)  FROM @my_stage(schema=(c1 int, c2 float, c3 string))
+)
+```
+
+if the same schema is used frequently, user can create table for it and use table name.
+
+```sql
+SELECT  c1, c2 + c1, trim(c2)  FROM @my_stage(schema='db_name.table_name')
+```
+
+and user can get knowledge about the data with `desc <table>` instead of `infer schema` each time.
```

Comment:
Does this mean that if a schema is not attached, we will fall back to inferring the schema instead?

Reply:
yes.